test/interpreter/function-pduration.spec.ts

```diff
@@ -0,0 +1,26 @@
+import {ErrorType, HyperFormula} from '../../src'
+import {ErrorMessage} from '../../src/error-message'
+import {adr, detailedError} from '../testUtils'
+
+describe('Function PDURATION', () => {
+  it('should return #NA! error with the wrong number of arguments', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=PDURATION(1,1)', '=PDURATION(1, 1, 1, 1)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+    expect(engine.getCellValue(adr('B1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+  })
+
+  it('should calculate the correct value with correct arguments and defaults', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=PDURATION(2%, 12, 24)'],
+      ['=PDURATION(2%, -12, -24)'],
+      ['=PDURATION(0, 1, 1)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toBeCloseTo(35.00278878, 6)
+    expect(engine.getCellValue(adr('A2'))).toEqual(detailedError(ErrorType.NUM, ErrorMessage.ValueSmall))
+    expect(engine.getCellValue(adr('A3'))).toEqual(detailedError(ErrorType.NUM, ErrorMessage.ValueSmall))
+  })
+})
```

Comment:
> Rate argument test missing for incorrect value. What if a negative value was passed? What if inncorect type was passed, ie. string? [2] "Constraints: Rate > 0; CurrentValue > 0; SpecifiedValue > 0"
> I don't see a test with #VALUE! error. But that condition is clearly described [1]: "If arguments are not using valid data types, PDURATION returns the #VALUE! error value."

That is because it is extremely simple function that fully uses our wrapper. Thus putting such test is testing the wrapper, not the implementation.

Reply:
👍